Projects/My_website/myWebsite/blog/views.py

```diff
@@ -18,4 +18,45 @@ def postDetail(request,slug):
     return render(request,"blog/postDetail.html",{'post':iPost,"post_tag":iPost.tags.all()})
 
 def createBlog(request):
-    return render(request,'blog/createBlog.html')
\ No newline at end of file
+    if request.method== 'POST':
+        title = request.POST['title']
+        excerpt = request.POST['excerpt']
+        content = request.POST['content']
+        tag = request.POST['tag']
+        image = request.POST['file']
+
+       
+        if request.user.is_authenticated:
+
+            first_name = request.user.first_name
+            last_name = request.user.last_name
+            email = request.user.email
+            author=None
+
+            if Author.objects.filter(email=email).exists():
+                author = Author.objects.filter(email=email).first()
```

Comment:
@codal-uchoudhary , no need to query two times here..pls optimise this.

Also, you have added the validation in signup that we cannot create a user with an email address that already exists in our DB. So instead of .filter(), we can use .get() here.

Reply:
User and Author both are different model ,
there is no relation between them.
> if no author found according to email, so i have to create a new author in the Author model using current user data.

Reply:
@codal-uchoudhary , so author table also must not have entries which have duplicate emails right ?